pkg/controllers/managementuser/rbac/prtb_handler_test.go

```diff
@@ -186,3 +192,273 @@ func Test_manager_checkForGlobalResourceRules(t *testing.T) {
+					return &rbacv1.ClusterRole{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "myrole-promoted",
+						},
+						Rules: []rbacv1.PolicyRule{
+							{
+								APIGroups:     []string{"another.cattle.io"},
+								ResourceNames: []string{"foobar"},
+								Resources:     []string{"nodes"},
+								Verbs:         []string{"create"},
+							},
+							{
+								APIGroups:     []string{"management.cattle.io"},
+								ResourceNames: []string{"local"},
+								Resources:     []string{"myresource"},
+								Verbs:         []string{"get", "list"},
+							},
+						},
+					}, nil
+				},
+			},
+			clusterRolesMock: &typesrbacv1fakes.ClusterRoleInterfaceMock{
+				CreateFunc: func(in1 *v1.ClusterRole) (*v1.ClusterRole, error) {
+					assert.Fail(t, "unexpected call to ClusterRole Create")
+					return in1, nil
+				},
+				UpdateFunc: func(in1 *v1.ClusterRole) (*v1.ClusterRole, error) {
+					assert.Equal(t, &rbacv1.ClusterRole{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "myrole-promoted",
+						},
+						Rules: []rbacv1.PolicyRule{
+							{
+								APIGroups:     []string{"another.cattle.io"},
+								ResourceNames: []string{"foobar"},
+								Resources:     []string{"nodes"},
+								Verbs:         []string{"create"},
+							},
+							{
+								APIGroups:     []string{"management.cattle.io"},
+								ResourceNames: []string{"local"},
+								Resources:     []string{"myresource"},
+								Verbs:         []string{"delete", "list"},
+							},
+						},
+					}, in1)
+					return nil, nil
+				},
+			},
+			want: "myrole-promoted",
+		},
+		{
+			name: "removing verbs from policy will remove the rule",
+			args: args{
+				rtName:   "myrole",
+				resource: "myresource",
+				newVerbs: sets.New[string](),
+				baseRule: rbacv1.PolicyRule{
+					APIGroups:     []string{"management.cattle.io"},
+					ResourceNames: []string{"local"},
+				},
+			},
+			crListerMock: &typesrbacv1fakes.ClusterRoleListerMock{
+				GetFunc: func(namespace, name string) (*v1.ClusterRole, error) {
+					return &rbacv1.ClusterRole{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "myrole-promoted",
+						},
+						Rules: []rbacv1.PolicyRule{
+							{
+								APIGroups:     []string{"another.cattle.io"},
+								ResourceNames: []string{"foobar"},
+								Resources:     []string{"nodes"},
+								Verbs:         []string{"create"},
+							},
+							{
+								APIGroups:     []string{"management.cattle.io"},
+								ResourceNames: []string{"local"},
+								Resources:     []string{"myresource"},
+								Verbs:         []string{"get", "list"},
+							},
+						},
+					}, nil
+				},
+			},
+			clusterRolesMock: &typesrbacv1fakes.ClusterRoleInterfaceMock{
+				CreateFunc: func(in1 *v1.ClusterRole) (*v1.ClusterRole, error) {
+					assert.Fail(t, "unexpected call to ClusterRole Create")
+					return in1, nil
+				},
+				UpdateFunc: func(in1 *v1.ClusterRole) (*v1.ClusterRole, error) {
+					assert.Equal(t, &rbacv1.ClusterRole{
```

Comment:
we don't check if this method is called, and maybe we could even check how many times was called? (just if it makes senses e.g it's called more than once). 
If update is never called this test would pass and this assert would never be verified.

Reply:
That's true. There are some `XXXCalls` methods in these mocks. I can probably add some more checks about this, and dropping the `assert.Fail` when we don't want the method being called.
I think that could be dropped anyway (test will panic if called)